app/components/message-box.js

```diff
@@ -21,6 +22,13 @@ export default Ember.Component.extend({
     }
   },
 
+  init() {
+    this._super(...arguments);
+    this.router.addObserver('currentRouteName', () => {
```

Comment:
How does this work if there is no `currentRouteName` variable in the file??

Reply:
`currentRouteName` is available because of router service.